modules/core/src/shaderlib/project/project-functions.ts

```diff
@@ -25,11 +32,23 @@ function lngLatZToWorldPosition(lngLatZ, viewport, offsetMode = false) {
   return p;
 }
 
-function normalizeParameters(opts) {
-  const normalizedParams = {...opts};
-
-  let {coordinateSystem} = opts;
-  const {viewport, coordinateOrigin, fromCoordinateSystem, fromCoordinateOrigin} = opts;
+function normalizeParameters(opts: {
+  viewport: Viewport;
+  coordinateSystem: typeof COORDINATE_SYSTEM[keyof typeof COORDINATE_SYSTEM];
```

Comment:
Nit: This type is pretty ugly. At some point it would be good to move to string constants and get rid of need to import the `COORDINATE_SYSTEM` object. I suppose that could be done in major release or be supported in parallel. 

We could also just enumerate the integer values here

```
coordinateSystem: 0 | 1 | 2 | 3,
```

And then start adding

```
coordinateSystem: 0 | 1 | 2 | 3 | 'meter-offsets' | 'lng-lat',
```